.github/workflows/tests.yml

```diff
@@ -305,14 +305,21 @@ jobs:
         # NB: mesa-utils is needed for PyQT testing (https://github.com/Microsoft/WSL/issues/1246#issuecomment-356425862)
         sudo apt update && sudo DEBIAN_FRONTEND=noninteractive apt install -y gcc git curl mesa-utils
     - uses: actions/checkout@v2
+    - name: Copy SCT from $GITHUB_WORKSPACE to ~
+      run: |
+        cd ..
+        cp -r spinalcordtoolbox ~/spinalcordtoolbox
```

Comment:
So, the reason you can't `defaults: working-directory: ~/` is because `working-directory:` is applied by the runner itself, before it shells out to the shell (defined by `defaults: shell:` or just the system default), and that means that in this case, you need to use Windows syntax, and can only pick Windows folders: `defaults: working-directory: C:\Users\runneradmin`.

And `actions/checkout` too: it's unaware that we've installed WSL and, being a self-contained "Action", uses its own scripts and shells, so it can *only* check out into Windows folders.

Hence to get this to work we would have to:

- a: patch `Vampire/setup-wsl` to add a `wsl-working-directory:` option to instruct `wsl-bash` to `cd` a second time once it the Github Runner launches it
- b. replace `actions/shell` with a direct `git clone` (run under `wsl-bash`)

In light of all that, your workarounds seem like the best option unfortunately.

I wish we had a good issue we could at least link to explain these odd lines, Maybe the writeup I just did at https://github.com/Vampire/setup-wsl/issues/12 but I don't know if it's clear enough.

Reply:
Thank you for digging into this and debugging how the working directory is set. When I first started fiddling with this, `setup-wsl` felt like an opaque thing to me, so I'm glad we have a better understanding of how things work now.

Reply:
https://github.com/Vampire/setup-wsl/discussions/13#discussioncomment-1641892 says we should be able to add

```
shell: wsl-bash
wsl-shell-command: bash -c "cd && bash --noprofile --norc -euo pipefail '{0}'"
```

to get it to behave; and we don't need to set `working-directory` or patch it to add `wsl-working-directory`, it'll just be in `$HOME` by default.

and also that my attempts to use `working-directory` *should* have worked, I was just using the wrong path (I don't think I was but I can check again).

That doesn't deal with `actions/checkout`, though.